es/es-server/src/main/java/org/elasticsearch/client/ClusterAdminClient.java

```diff
@@ -113,4 +116,15 @@
      * that update the cluster state (for example, a create index operation)
      */
     PendingClusterTasksRequestBuilder preparePendingClusterTasks();
+
+    /**
+     * Nodes stats of the cluster.
+     *
+     * @param request  The nodes info request
+     * @param listener A listener to be notified with a result
+     * @see org.elasticsearch.client.Requests#nodesStatsRequest(String...)
+     */
+    void nodesStats(NodesStatsRequest request, ActionListener<NodesStatsResponse> listener);
```

Comment:
This is the api part which is necessary to have to be called later from https://github.com/elastic/elasticsearch/blob/70b8d7bc64f165735502de9d8c5fa673fa21e02b/server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java#L248
to fix the ClusterInfo.